app/scripts/lib/snap-keyring/metrics.test.ts

```diff
@@ -0,0 +1,99 @@
+import getSnapAndHardwareInfoForMetrics from './metrics';
+
+describe('getSnapAndHardwareInfoForMetrics', () => {
+  let getSelectedAddress: jest.Mock;
+  let getAccountType: jest.Mock;
+  let getDeviceModel: jest.Mock;
+  let messenger;
+
+  beforeEach(() => {
+    getSelectedAddress = jest.fn();
+    getAccountType = jest.fn();
+    getDeviceModel = jest.fn();
+    messenger = {
+      call: jest.fn(),
+    };
+  });
```

Comment:
need to reset the mocks after each call

Reply:
Done in [41e0b27](https://github.com/MetaMask/metamask-extension/pull/21347/commits/41e0b273c52458b30403a89dbed10598f719f12b)